src/gc-tls-third-party.h

```diff
@@ -0,0 +1,12 @@
+// This file is a part of Julia. License is MIT: https://julialang.org/license
+
+// Pick the appropriate third-party implementation
+#ifdef WITH_THIRD_PARTY_HEAP
+#if WITH_THIRD_PARTY_HEAP == 1
+#include "gc-tls-mmtk.h"
+#endif
+// To be extended by other GC implementations
+// #if WITH_THIRD_PARTY_HEAP == X
+//      #include gc-tls-XXXX.h
+// #endif
+#endif
```

Comment:
Having this file seems a bit too much at the moment.

We don't plan to support any other third-party GC in the foreseeable future, so I'd just directly inline this logic:

```C
#ifdef WITH_THIRD_PARTY_HEAP
#if WITH_THIRD_PARTY_HEAP == 1
#include "gc-tls-mmtk.h"
#endif
```

into `julia_threads.h`.